DigitalLearningSolutions.Data/Services/GroupsService.cs

```diff
@@ -371,7 +373,7 @@ public void UpdateGroupName(int groupId, int centreId, string groupName)
             );
 
             var groupDelegates = GetGroupDelegates(groupId);
-            var groupCourse = groupsDataService.GetGroupCourseById(groupCustomisationId)!;
+            var groupCourse = groupsDataService.GetGroupCourseForCentre(groupCustomisationId, centreId)!;

```

Comment:
I'm not really sure how we can handle this better, we're selecting the record we just inserted. If this is null the insert must have failed.

Reply:
Well not quite - the new code assumes that the course you've added is visible to the centre. Currently this method adds the course to the group either way, then will only return a non-null value if the course is visible to the centre. If we get a `null` back I reckon we should roll back the transaction, ideally log what's happened, and throw an exception. The transaction rollback and exception probably happen already, but it would be nice to make it a bit more obvious.